src/rm_topk.c

```diff
@@ -187,25 +187,40 @@ static int TopK_Count_Cmd(RedisModuleCtx *ctx, RedisModuleString **argv, int arg
 }
 
 static int TopK_List_Cmd(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
-
-    if (argc != 2)
+    if (argc != 2 && argc != 3)
         return RedisModule_WrongArity(ctx);
 
+    bool withCount = 0;
+    if (argc == 3) {
+        size_t len;
+        const char *countStr = RedisModule_StringPtrLen(argv[2], &len);
+        if (strncasecmp(countStr, "WITHCOUNT", len) == 0) {
+            withCount = 1;
+        } else {
+            return RedisModule_ReplyWithError(ctx, "WITHCOUNT keyword expected");
+        }
+    }
+
     TopK *topk = NULL;
     if (GetTopKKey(ctx, argv[1], &topk, REDISMODULE_READ) != REDISMODULE_OK) {
         return REDISMODULE_OK;
     }
-    uint32_t k = topk->k;
-    char **heapList = TOPK_CALLOC(k, (sizeof(char *)));
-    TopK_List(topk, heapList);
-    RedisModule_ReplyWithArray(ctx, k);
-    for (int i = 0; i < k; ++i) {
-        if (heapList[i] != NULL) {
-            RedisModule_ReplyWithSimpleString(ctx, heapList[i]);
+    HeapBucket *heapList = TopK_List(topk);
+    RedisModule_ReplyWithArray(ctx, REDISMODULE_POSTPONED_ARRAY_LEN);
```

Comment:
@ashtul given we know there are implications on `REDISMODULE_POSTPONED_ARRAY_LEN` and that we know beforehand the size of the array ( either k or 2xk ) lets keep RedisModule_ReplyWithArray with the proper arraylen. 
agree?

Reply:
We might break with less than `k` elements if the heap is not full yet.

Reply:
@filipecosta90 a single use of `REDISMODULE_POSTPONED_ARRAY_LEN` would not hurt performance. From what I know, we had an issue in search where we had wide use of it in a reply true of possibly millions of results.
The other way to do it is to iterate twice over the array and count the legal result but this seems excessive to me. WDYT?